lib/coinbase/address.rb

```diff
@@ -14,10 +14,12 @@ module Coinbase
   class Address
     # Returns a new Address object. Do not use this method directly. Instead, use Wallet#create_address.
     # @param model [Coinbase::Client::Address] The underlying Address object
+    # @param addresses_api [Coinbase::Client::AddressesApi] The Addresses API object
     # @param key [Eth::Key] The key backing the Address
     # @param client [Jimson::Client] (Optional) The JSON RPC client to use for interacting with the Network
-    def initialize(model, key, client: Jimson::Client.new(Coinbase.base_sepolia_rpc_url))
+    def initialize(model, addresses_api, key, client: Jimson::Client.new(Coinbase.base_sepolia_rpc_url))
       @model = model
+      @addresses_api = addresses_api
       @key = key
       @client = client
```

Comment:
We probably want `attr_reader`s for these since they should only be set in the context of the initialize method